api/src/test/java/org/apache/iceberg/transforms/TestTruncate.java

```diff
@@ -75,6 +75,8 @@ public void testTruncateString() {
         "abcde", trunc.apply("abcdefg"));
     Assert.assertEquals("Should not pad strings shorter than length",
         "abc", trunc.apply("abc"));
+    Assert.assertEquals("Should not alter strings equal to length",
+        "abcde", trunc.apply("abcde"));
```

Comment:
Not exactly related so happy to remove if anybody rejects (or move it into its own PR).